### PR TITLE
Add support for most `random` cheatcodes

### PR DESCRIPTION
**Pull request by palinatolmach**
This PR partially addresses  https://github.com/runtimeverification/kontrol/issues/876.

This PR adds support for the following cheatcodes recently added to `forge-std`, and adds `end-to-end` tests for them:

```solidity
    /// Returns a random `address`.
    function randomAddress() external returns (address);

    /// Returns a random `bool`.
    function randomBool() external view returns (bool);

    /// Returns a random byte array value of the given length.
    function randomBytes(uint256 len) external view returns (bytes memory);

    /// Returns a random fixed-size byte array of length 4.
    function randomBytes4() external view returns (bytes4);

    /// Returns a random fixed-size byte array of length 8.
    function randomBytes8() external view returns (bytes8);

    /// Returns a random uint256 value.
    function randomUint() external returns (uint256);

    /// Returns a random `uint256` value of given bits.
    function randomUint(uint256 bits) external view returns (uint256);

    /// Returns random uint256 value between the provided range (=min..=max).
    function randomUint(uint256 min, uint256 max) external returns (uint256);
```

The following cheatcodes will be added in a follow up PR; they're currently causing branching due to some signed ints specifics:
```solidity
    /// Returns a random `int256` value.
    function randomInt() external view returns (int256);

    /// Returns a random `int256` value of given bits.
    function randomInt(uint256 bits) external view returns (int256);
```

### CODE REVIEW

File: src/kontrol/kdist/cheatcodes.md [hunk 1]
```diff
@@ -378,20 +378,63 @@ This rule returns a symbolic integer of up to the bit width that was sent as an
...
+function randomUint() external returns (uint256);
+function randomUint(uint256) external returns (uint256);
+function randomUint(uint256,uint256) external returns (uint256);
+```
+
+`cheatcode.call.randomUint` will match when the `randomUint` cheat code function is called.
+This rule returns a symbolic integer of up to the bit width that was sent as an argument.
+
+```{.k .symbolic}
+    rule [cheatcode.call.randomUint]:
```
Comment (anvacaru) [inline src/kontrol/kdist/cheatcodes.md:393]: The naming of the rule seems a bit confusing here, as this cheat code is the one that takes the width argument.

File: src/kontrol/kdist/cheatcodes.md [hunk 2]
```diff
@@ -378,20 +378,63 @@ This rule returns a symbolic integer of up to the bit width that was sent as an
...
-`cheatcode.call.freshBool` will match when the `freshBool` cheat code function is called.
+`cheatcode.call.freshBool` will match when the `freshBool` or `randomBool` cheat code function is called.
 This rule returns a symbolic boolean value being either 0 (false) or 1 (true).
 
 ```{.k .symbolic}
     rule [cheatcode.call.freshBool]:
          <k> #cheatcode_call SELECTOR _ => .K ... </k>
          <output> _ => #buf(32, ?WORD) </output>
       requires SELECTOR ==Int selector ( "freshBool()" )
+         orBool SELECTOR ==Int selector ( "randomBool()" )
```
Comment (anvacaru) [inline src/kontrol/kdist/cheatcodes.md:437]:
```suggestion
        orBool SELECTOR ==Int selector ( "randomBool()" )
```

File: src/kontrol/kdist/cheatcodes.md [hunk 3]
```diff
@@ -413,24 +459,53 @@ This rule returns a fully symbolic byte array value of the given length.
...
-`foundry.call.freshAddress` will match when the `freshAddress` cheat code function is called.
+`foundry.call.freshAddress` will match when the `freshAddress` or `randomAddress` cheat code function is called.
 This rule returns a symbolic address value.
 
 ```{.k .symbolic}
     rule [foundry.call.freshAddress]:
          <k> #cheatcode_call SELECTOR _ => .K ... </k>
          <output> _ => #buf(32, ?WORD) </output>
       requires SELECTOR ==Int selector ( "freshAddress()" )
+         orBool SELECTOR ==Int selector ( "randomAddress()" )
```
Comment (anvacaru) [inline src/kontrol/kdist/cheatcodes.md:508]:
```suggestion
        orBool SELECTOR ==Int selector ( "randomAddress()" )
```

**Review by anvacaru — APPROVED** [review]

File: src/kontrol/kdist/cheatcodes.md (hunk 2 above)
Comment (palinatolmach) [reply src/kontrol/kdist/cheatcodes.md:437]: Fix whitespaces in review

File: src/kontrol/kdist/cheatcodes.md (hunk 1 above)
Comment (palinatolmach) [reply src/kontrol/kdist/cheatcodes.md]: Thanks! Fixed in https://github.com/runtimeverification/kontrol/pull/877/commits/226e3f5fa9c94e60986c7d945a69caffb52aca16.

**Pull request merged**